### ISSUE TITLE
UCP: Migrate `dataForKeyColumnUsage` from package `infoschema` to `executor`

### ISSUE THREAD

**Issue by lonng** [pingcap/tidb#15041]
## Description

This is sub-task of https://github.com/pingcap/tidb/issues/14828

The package `infoschema` should be used to store meta-information for virtual tables only.
But there is some retrieval data logic of the virtual table, which messes the `infoschema` .
PR #14704 has begun to migrate them. You can reference #14704 to help you migrate the `dataForKeyColumnUsage`

## Score 

* 50

## Mentor(s)

* @lonng 

## Recommended Skills

* Go

## Learning Materials

- PR: #14704

**Comment by zz-jason** [comment]
> @lonng is this not part of the UCP?

Hi @psinghal20 sorry for the confusion. This is a PCP issue. Don't worry, just ignore the error message above. Once your PR is merged, you can get the score of this issue.

**Comment by you06** [comment]
@psinghal20 try picking again please.
